file-format-spec.ts

```diff
@@ -0,0 +1,87 @@
+// This file contains types which specify the speedscope file format.
```

Comment:
This is definitely a simpler format to create.  People who want to export to this format don't need to recreate the call tree anymore.  It definitely works with start/stop types of trace events.

How does a set of raw stack samples work with this format now?  I'm guessing your planning to export instruments files as start/stop spans vs a list of stack samples?  Or extend this later to export instruments-style stack samples?

Reply:
> I'm guessing your planning to export instruments files as start/stop spans vs a list of stack samples?

Yeah, that was the idea. Anything currently imported into speedscope can be safely exported in this way right now, though it's perhaps a bit misleading since the actual input was sample-based, not event based.

An alternative approach would be to support both the original format I proposed and this new format. The door for doing that in the future would still be open, even if that isn't done in the initial version. That was the purpose of making `IProfile` a discriminated union of types.

In this current format, would this be useful for your immediate needs? The code in this PR both supports importing this format if you want to write tools that can output profiles to import into speedscope and supports exporting to this format if you wanted to use speedscope to e.g. import instruments profiles and export to an easier to parse format.

Reply:
Yeah this would work great.

Thank you so much for doing all of this!